subsys/mgmt/ec_host_cmd/backends/ec_host_cmd_backend_espi.c

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2022 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/espi.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/mgmt/ec_host_cmd/ec_host_cmd.h>
+#include <zephyr/mgmt/ec_host_cmd/backend.h>
+
+LOG_MODULE_REGISTER(host_cmd_espi, CONFIG_EC_HC_LOG_LEVEL);
+
+#define RX_HEADER_SIZE (sizeof(struct ec_host_cmd_request_header))
+
+struct ec_host_cmd_espi {
+	/* eSPI device instance */
+	const struct device *espi_dev;
+	/* Context for read operation */
+	struct ec_host_cmd_rx_ctx *rx_ctx;
+	/* Transmit buffer */
+	struct ec_host_cmd_tx_buf *tx;
+	/* eSPI callback */
+	struct espi_callback espi_cb;
+};
+
+#define EC_HOST_CMD_ESPI_DEFINE(_name)                                                             \
```

Comment:
I'm not sure if it's a good idea to dereference non-existing yet object, even if it's inside macro that will be used later. If you remove the API requirement to know specific drivers, it won't be needed.

Reply:
Also I don't see the need for this macro, if it's only used once with one name

Reply:
I think it is not a problem if it is within one file.

Reply:
I think using macro is a good approach when you create a few variables that depend on their names.